statsmodels/tsa/statespace/dynamic_regression.py

```diff
@@ -0,0 +1,321 @@
+        return selection
+
+    def update(self, params, transformed=True, **kwargs):
+        """
+        Update the parameters of the model
+
+        Updates the representation matrices to fill in the new parameter
+        values.
+
+        Parameters
+        ----------
+        params : array_like
+            Array of new parameters.
+        transformed : boolean, optional
+            Whether or not `params` is already transformed. If set to False,
+            `transform_params` is called. Default is True..
+
+        Returns
+        -------
+        params : array_like
+            Array of parameters.
+        """
+        obs_cov = params[0]
+        state_cov_params = params[1:]
+        self.ssm.obs_cov[0,0,0]= obs_cov
+        self.ssm.state_cov[:,:,0][np.diag_indices(
+            self.k_state_cov)] = state_cov_params
+        
+
+    def transform_params(self, unconstrained):
+        """
+        Transform unconstrained parameters used by the optimizer to constrained
+        parameters used in likelihood evaluation.
+
+        Used primarily to enforce stationarity of the autoregressive lag
+        polynomial, invertibility of the moving average lag polynomial, and
+        positive variance parameters.
+
+        Parameters
+        ----------
+        unconstrained : array_like
+            Unconstrained parameters used by the optimizer.
+
+        Returns
+        -------
+        constrained : array_like
+            Constrained parameters used in likelihood evaluation.
+        """
+        unconstrained = np.array(unconstrained, ndmin=1)
+        constrained = np.zeros(unconstrained.shape, unconstrained.dtype)
+        start = 0
+        # Transform the covariance params to be positive
+        constrained[start:] = unconstrained[start:]**2
+        return constrained
+    
+    def untransform_params(self, constrained):
+        """
+        Transform constrained parameters used in likelihood evaluation
+        to unconstrained parameters used by the optimizer
+
+        Used primarily to reverse enforcement of stationarity of the
+        autoregressive lag polynomial and invertibility of the moving average
+        lag polynomial.
+
+        Parameters
+        ----------
+        constrained : array_like
+            Constrained parameters used in likelihood evaluation.
+
+        Returns
+        -------
+        constrained : array_like
+            Unconstrained parameters used by the optimizer.
+        """
+        constrained = np.array(constrained, ndmin=1)
+        unconstrained = np.zeros(constrained.shape, constrained.dtype)
+        # Transform the AR parameters (phi) to be stationary
+        start = 0
+        # Untransform the covariance params which have been transformed
+        # to be positive
+        unconstrained[start:] = np.sqrt(constrained[start:])
+        return unconstrained
+
+def plot_dynamic_regression(results,which="smoothed",figsize=None,
```

Comment:
Need spaces.  Should run

`flake8 --isolated statsmodels/tsa/statespace/dynamic_regression.py`

and fix any issues.